File: app/src/main/java/com/poh/game/Game.java
package com.poh.game;

import com.poh.playObjects.Square;

import java.util.concurrent.atomic.AtomicInteger;

public abstract class Game {
    protected int duration = 15;
    protected int frequency = 2;
    protected AtomicInteger score;
    public static final int VERTICAL_SIZE = 5;
    public static final int HORIZONTAL_SIZE = 5;
    protected Class[] possiblePlayObjects;

    public abstract void play();
}

File: app/src/main/java/com/sadwyn/numbers/game/Game.kt
package com.sadwyn.numbers.game

import com.sadwyn.numbers.objects.Square
import java.util.concurrent.atomic.AtomicInteger

abstract class Game(val gameEventListener: GameEventListener) {
    protected var frequency: Long = 1000
    protected lateinit var score: AtomicInteger
    protected var grid: Array<Square?> = Array(25) {null}

    abstract fun updateScore(value: Int)
    abstract fun getScore() : Int
    abstract fun play()
    abstract fun pause()
    abstract fun resume()
}
